src/main/java/graphql/nadel/engine/PathMapper.java

```diff
@@ -19,21 +18,31 @@ public ExecutionPath mapPath(ExecutionPath executionPath, String resultKey, Unap
             //
             // /issue/reporter might lead to /userById and hence we need to collapse the top level hydrated field INTO the target field
             List<Object> tmp = environment.parentNode.getExecutionPath().toList();
-            tmp.add(fieldSegments.get(fieldSegments.size() - 1));
-            fieldSegments = tmp;
+            tmp.add(pathSegments.get(pathSegments.size() - 1));
+            pathSegments = tmp;
+        } else {
```

Comment:
good catch

Reply:
this look nearly identically to the hydration part now: if the current path only contains one more elements than the environment.parentPath they are identically. Can we try to make them the same?

Reply:
Done @andimarek 

I was originally worried there might be more than one child segment to add so I did the sublist, but this seems to work.